.github/workflows/build-images.yml

```diff
@@ -297,7 +301,11 @@ jobs:
           rm -rf "scripts/ci"
           mv "main-airflow/scripts/ci" "scripts"
       - name: "Free space"
-        run: ./scripts/ci/tools/free_space.sh
+        run: ./scripts/ci/tools/freespace.py
```

Comment:
Hmm. I think also this way it won't work. I think you cannot combine "run" and "setup-python" actions. Those shoudl be separate steps in the job:

Those two shoudl be run once at the beginning of each job:

1) setup_python (with pip cache)
2) run: pip install "./dev/breeze" - to install the dependenices 

Then the "free_space.py" (and any future steps like this) could be run as the next steps in the same job.

* run: free_space.py 
* run: build_image.py (later) 
* ....

This way the "initial" steps are setting up python virtualenv (with caching) and install the right dependencies (such as rich) once and all the follow-up run methods will have the "python" virtual environment prepared and with the right depenencies installed (and the "pip" caching of setup-python is really nice because it will keep the dependencies in a cache which means that in most cases this environment creation as initial steps of the job will be very fast).

Reply:
I see, I am checking the documentation about YAML syntax for workflow
https://docs.github.com/es/actions/learn-github-actions/workflow-syntax-for-github-actions. I hope to fix it soon. Thanks!

Reply:
@potiuk I am adding setup-python before free space (first code below). In some jobs, I found other ways to set up python. for example (see second code below), which one I should consider?

```
 - uses: actions/setup-python@v2
        with:
          python-version: '3.7'
          cache: 'pip'
      - run: pip install .
      - name: "Free space"
        run: ./scripts/ci/tools/freespace.py
```

```
 - name: "Setup python"
        uses: actions/setup-python@v2
        with:
          python-version: ${{needs.build-info.outputs.defaultPythonVersion}}
```

Reply:
I think we should aim to install everywhere python with dependencies from 'dev/breeze". That was not needed  before, but if move all the "python" scripts we run in CI to "dev/breeze", then it will be easier to rely on whatever dependencies we have there (for example rich). 

So I think we should really aim to have this everywhere:

```
 - uses: actions/setup-python@v2
    with:
      python-version: '3.7'
      cache: 'pip'
  - run: pip install ./dev/breeze
```

Unfortunately, we will have to repeat it everywhere as there is no easy way currently to reuse parts of the jobs easily (as far as I know).

Actually, you made me think a bit and I think we could do even better (that would be really nice!):

1) you could move the "freespace.py" to "./dev/breeze/ci/freespace.py"

2) we could add freespace.py as as "entrypoint"

See here: https://github.com/apache/airflow/blob/4dd751d2fc3f091417146079d7d1b3ac6dd86c9c/dev/breeze/setup.cfg#L64

We could do something like:
```
console_scripts=
    Breeze2=airflow_breeze.breeze:main
    freespace=ci.freespace.main
```

And that would be really nice, because after running:
```
pip install ./dev/breeze
```

the `freespace` script will become available on the PATH and you will be able to call it just like:

```
run: freespace
```

That woudl be really, really nice in the `pipx` scenario (@uranusjr !) which i start liking more and more, because in case of `pipx` installation, freespace will not ony become available in path but also will have autocomplete enabled after we add it to the autocomplete scripts. Maybe freespace is not best example but we have more "standalone" scripts that are not part of `breeze`  that we could expose in similar way eventually.

Reply:
yay, let me check how it could be with an entry_points :)

Reply:
We are calling setup.cfg in several parts of the code, I am a little confused with this.
setup.py is for python dependencies, and, for what kind of configurations do we have on setup.cfg?

Reply:
@potiuk since in setup.cfg has: package_dir in src, so I moved freespace.py from: dev/breeze/ci to src/airflow_breeze/ci and I added this entry[point:

```
[options.entry_points]
console_scripts=
    Breeze2=airflow_breeze.breeze:main
    freespace=airflow_breeze.ci.freespace:main
```

Is it not being recognized. It may be omitting one step.